### PR TITLE
Add alternative Harvester: Kaizen

### PR DESCRIPTION
**Pull request by yukiisbored**
Kaizen is an alternative approach to do Harvesting, it focuses on query and information collecting to generate more queries. Kaizen also uses Twitter API to get trending topics to increase the chance of getting relevant or wanted messages (This feature is optional).

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by mariobehling** [comment]
Thanks. Is there an issue with some more background info? Please help everyone to follow the development and always open an issue with some info first (that could be searched in the future as well) and match each PR with an issue.

**Comment by mariobehling** [comment]
Could you ensure, that all settings/configurations can be accessed through the settings page as well, please? E.g. that the user can enter API and Twitter credentials?

**Comment by yukiisbored** [comment]
@mariobehling I think the twitter credentials variables can't be viewed from the settings page because they're commented out of `configuration.properties`

**Comment by mariobehling** [comment]
@yukiisbored Would probably be good to have a settings/config page that can only be viewed and edited by admins. Then we can have all of it here.

**Comment by mariobehling** [comment]
We should also have an option to switch different harvesters on/off. 

How do you want to proceed? Should we merge this and keep the other things on the list of to dos?

**Comment by yukiisbored** [comment]
@mariobehling I've already added an option to switch between two containers, Currently we can only have one type of harvester since Caretaker spawns a number of them.

**Comment by yukiisbored** [comment]
For the settings page, I'll do that on a different PR.

**Comment by yukiisbored** [comment]
@mariobehling I've created the issue and mentioned it in the commit message for c1c0743. For the settings page and other settings-related stuff, it's better if we do it on another PR.

**Comment by yukiisbored** [comment]
@mariobehling The codacy issue is out of my reach because it's caused by the harvest method of the original harvester

File: src/org/loklak/harvester/PushThread.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                try {Thread.sleep((i + 1) * 3000);} catch (InterruptedException e1) {}
+            }
+        }
+        String q = tl.getQuery();
+        tl.setQuery(null);
+        DAO.outgoingMessages.transmitTimelineToBackend(tl);
+        DAO.log("retrieval of " + tl.size() + " new messages for q = " + q + ", scheduled push");
+    }
+
+}
```
Comment (jayvdb) [inline src/org/loklak/harvester/PushThread.java]: Missing EOL at EOF ;-)

**Pull request merged**